src/dmd/expressionsem.d

```diff
@@ -6809,48 +6809,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             if (tb.ty == Tstruct)
             {
                 ad = (cast(TypeStruct)tb).sym;
-                auto f = ad.aggDelete;
-                auto fd = ad.dtor;
-                if (!f)
-                {
-                    semanticTypeInfo(sc, tb);
-                    break;
-                }
```

Comment:
This code was a bit surprising: If `ad.aggDelete` is `false` (and it should, because otherwise we error out), the `if` is always followed and the code that follows is never executed. However the code seemed to account for the possibility of `f` or `fd` being `null`, which doesn't make sense as `fd` can never be `null`. Maybe it's just historical baggage.